jmxtrans-output/jmxtrans-output-core/src/main/java/com/googlecode/jmxtrans/model/output/support/influxdb/InfluxDBMessageFormatter.java

```diff
@@ -0,0 +1,278 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.googlecode.jmxtrans.model.output.support.influxdb;
+
+import com.google.common.base.Function;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.googlecode.jmxtrans.exceptions.LifecycleException;
+import com.googlecode.jmxtrans.model.NamingStrategy;
+import com.googlecode.jmxtrans.model.Result;
+import com.googlecode.jmxtrans.model.naming.ClassAttributeNamingStrategy;
+import com.googlecode.jmxtrans.model.naming.JexlNamingStrategy;
+import com.googlecode.jmxtrans.model.naming.typename.TypeNameValue;
+import com.googlecode.jmxtrans.model.naming.typename.TypeNameValuesStringBuilder;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+import org.apache.commons.jexl2.JexlException;
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.net.UnknownHostException;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.collect.FluentIterable.from;
+import static com.googlecode.jmxtrans.util.NumberUtils.isNumeric;
+
+@EqualsAndHashCode
+@ToString
+public class InfluxDBMessageFormatter {
+
+	private static final Logger log = LoggerFactory.getLogger(InfluxDBMessageFormatter.class);
+	public static final String DEFAULT_TAG_NAME = "type";
+
+	private final ImmutableList<String> typeNames;
+	private final ImmutableMap<String, String> tags;
+	private final String tagName;
+	private final NamingStrategy metricNameStrategy;
+
+
+	private final boolean mergeTypeNamesTags;
+	private final boolean allKeysAsTags;
+	private final String hostnameTag;
+
+	public InfluxDBMessageFormatter(@Nonnull ImmutableList<String> typeNames,
+									@Nonnull ImmutableMap<String, String> tags) throws LifecycleException, UnknownHostException {
+		this(typeNames, tags, DEFAULT_TAG_NAME, null, true, false, "localhost");
+	}
+
+	public InfluxDBMessageFormatter(@Nonnull ImmutableList<String> typeNames,
+									@Nonnull ImmutableMap<String, String> tags,
+									@Nonnull String tagName,
+									@Nullable String metricNamingExpression,
+									boolean mergeTypeNamesTags,
+									boolean allKeysAsTags,
+									@Nullable String hostnameTag) throws UnknownHostException, LifecycleException {
+		this.typeNames = typeNames;
+		this.tags = tags;
+		this.tagName = tagName;
+		if (metricNamingExpression != null) {
```

Comment:
The construction of the metricNameStrategy should probably be done in the Factory. Let's try to keep constructors as simple as possible.